source/ota.c

```diff
@@ -86,182 +95,386 @@ typedef struct OtaStateTableEntry
                                        JsonDocModel_t * pDocModel );
 
-/* Decode the signature key from the job document and store it.*/
-
+/**
+ * @brief Decode the base64 encoded file signature key from the job document and store it in file context.
+ *
+ * @param pValueInJson Pointer to the value of the key in JSON buffer.
+ * @param valueLength Length of the value.
+ * @param pParamAdd Pointer to the location where the value is stored.
+ * @return DocParseErr_t DocParseErrNone if successful, JSON document parser errors.
+ */
 static DocParseErr_t decodeAndStoreKey( const char * pValueInJson,
                                         size_t valueLength,
                                         void * pParamAdd );
 
-/* Extract the value from json and store it into the allocated memory. */
-
+/**
+ * @brief Extract the value from json and store it into the allocated memory.
+ *
+ * @param pKey Name of the Key to extract.
+ * @param pValueInJson Pointer to the value of the key in JSON buffer.
+ * @param valueLength Length of the value.
+ * @param pParamAdd Pointer to the location where the value is stored.
+ * @param pParamSizeAdd Size required to store the param.
+ * @return DocParseErr_t DocParseErrNone if successful, JSON document parser errors.
+ */
 static DocParseErr_t extractAndStoreArray( const char * pKey,
                                            const char * pValueInJson,
                                            size_t valueLength,
                                            void * pParamAdd,
                                            uint32_t * pParamSizeAdd );
 
-/* Check if all the required parameters for the job are present in the JSON. */
-
+/**
+ * @brief Check if all the required parameters for job document are extracted from the JSON.
+ *
+ * @param pModelParam Structure to store the details of keys and where to store them.
+ * @param pDocModel Details of expected parameters in the job doc.
+ * @return DocParseErr_t DocParseErrNone if successful, JSON document parser errors.
+ */
 static DocParseErr_t verifyRequiredParamsExtracted( const JsonDocParam_t * pModelParam,
                                                     const JsonDocModel_t * pDocModel );
 
-/* Validate the version of the update received. */
-
+/**
+ * @brief Validate the version of the update received.
+ *
+ * @param pFileContext Information of file to be streamed.
+ * @return OtaErr_t
+ */
 static OtaErr_t validateUpdateVersion( const OtaFileContext_t * pFileContext );
 
-/* Check if the JSON can be parsed through a custom callback if initial parsing fails. */
-
+/**
+ * @brief Check if the JSON can be parsed through a custom callback if initial parsing fails.
+ *
+ * @param pJson JSON job document.
+ * @param messageLength Length of the job document.
+ * @param pFileContext Information of file to be streamed.
+ * @param pFinalFile File that stores all extracted params.
+ * @return OtaJobParseErr_t
```

Comment:
Maybe somthing more descriptive?

Reply:
Addressed in https://github.com/aws/ota-for-aws-iot-embedded-sdk/pull/116/commits/3e5b1d96211d212a79fd5aa4a1bd33a51add1e85 to add error descriptions